### DISCUSSION TITLE
GSMA OGW Product WS - Subscription/notification mechanism for QoD formal request

### DISCUSSION THREAD

**Discussion by jgarciahospital** [camaraproject/QualityOnDemand#189] [General]
Formal inclusion of the GSMA OGW Product WS request to include Subscription/notification mechanism as part of OGW services.

https://github.com/camaraproject/Commonalities/issues/8#issuecomment-1642419272

This request only formalises the inclusion of this service notifications/buscriptions mechanisms as part of the Open Gateway initiative, no changes are expected. The formal request includes the confiormation of the Notifications mechanisms and its alignment with Commonalities, apart from  examples or proposal for input-outputs of such mechanism, to be discussed and aligned with the current service API.

Additional context
More details included in the formal request file, attached
[Camara Submission - Subscription_Notifications v2.docx](https://github.com/camaraproject/QualityOnDemand/files/12123620/Camara.Submission.-.Subscription_Notifications.v2.docx)

**Comment by hdamker** [comment]
@jgarciahospital I actually don't know what is expected here ... just the note from my side, that what is requested in the document for the QoD example is what the guideline calls an "implicit" subscription. So there is - at the current time - no explicit subscription endpoint (and not needed for the functionality asked for).

What we have encountered during preparing v0.9.0 is that the current guidelines are not fully fitting for implicit subscriptions, at least they are not efficient, cf the discussion in https://github.com/camaraproject/QualityOnDemand/pull/177 and my comment in https://github.com/camaraproject/Commonalities/issues/8#issuecomment-1647512694

**Reply by jgarciahospital** [reply]
Not much, only oficializying that this feature is considered in OGW, as far as it applies to the proper API. If it's already agreed to move on with "explicit" model subscription and, on the other side, the model of notifications is already considering the use cases, not much to do then. Thanks Herbert for the response.

**Comment by hdamker** [comment]
Quality on Demand has only implicit notifications, no subscriptions.
